sherpa/bin/streaming_pruned_transducer_statelessX/beam_search.py

```diff
@@ -0,0 +1,296 @@
+from typing import List
+
+import k2
+import torch
+from decode import Stream
+from streaming_server import StreamingServer
+
+from sherpa import fast_beam_search_one_best, streaming_greedy_search
+
+
+class FastBeamSearch:
+    def __init__(
+        self, vocab_size, context_size, beam, max_states, max_contexts, device
+    ):
+        """
+        Args:
+          vocab_size:
+            Vocabularize of the BPE
+          context_size:
+            Context size of the RNN-T decoder model.
+          beam:
+            The beam for fast_beam_search decoding.
+          max_states:
+            The max_states for fast_beam_search decoding.
+          max_contexts:
+            The max_contexts for fast_beam_search decoding.
+          device:
+            Device on which the computation will occur
+        """
+        self.rnnt_decoding_config = k2.RnntDecodingConfig(
+            vocab_size=vocab_size,
+            decoder_history_len=context_size,
+            beam=beam,
+            max_states=max_states,
+            max_contexts=max_contexts,
+        )
+        self.decoding_graph = k2.trivial_graph(vocab_size - 1, device)
+        self.device = device
+        self.context_size = context_size
+
+    def get_attribute_stream(self, stream: "Stream"):
```

Comment:
This function initializes the stream for beam search.

Shall we rename it `init_stream`?